inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -126,24 +139,79 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const In
-        // temporarily switch back to plugin unroller from NGraph unroller until TI output names are corrected
-        // ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
-        ti_manager.run_passes(nGraphFunc);
+#ifndef USE_CNNNETWORK_LPT
+        bool enableInt8;
+#endif
+
+        {
+            // Note: instead of running all Conversion Transformations you can make up your own transformation pipeline
+            ngraph::pass::Manager manager;
+            manager.register_pass<ngraph::pass::InitNodeInfo>();
+            // WA: ConvertPriorBox must be executed before the 1st ConstantFolding pass
+            manager.register_pass<ngraph::pass::ConvertPriorBox>();
+            manager.register_pass<ngraph::pass::CommonOptimizations>();
+            manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
+            manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+
+
+            manager.set_callback(transformations_callback);
+            manager.run_passes(nGraphFunc);
+
+            const auto fp16_callback = [&baselineIsFP16](const std::shared_ptr<const ::ngraph::Node> &node) -> bool {
+                if (!baselineIsFP16 && node->get_output_element_type(0) == ngraph::element::f16) {
+                    baselineIsFP16 = true;
+                }
+
+                return true;
+            };
+
+            ngraph::pass::Manager conversion_manager;
+
+#ifndef USE_CNNNETWORK_LPT
+            enableInt8 = config.enableInt8 && ngraph::pass::low_precision::LowPrecisionTransformer::isFunctionQuantized(nGraphFunc);
+            if (enableInt8) {
+                // [WA part1] Convert quantized FP16 model to FP32 to avoid possible overflow and mixed precision errors
+                conversion_manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::f16, ngraph::element::f32);
+            }
+#endif
+
+            conversion_manager.set_callback(fp16_callback);
+            conversion_manager.run_passes(nGraphFunc);
+
+            ngraph::pass::Manager ti_manager;
+            // Unroll will be called after all conversions
+            // temporarily switch back to plugin unroller from NGraph unroller until TI output names are corrected
+            // ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
+            ti_manager.run_passes(nGraphFunc);
+        }
+
+#ifndef USE_CNNNETWORK_LPT
+        using namespace ngraph::pass::low_precision;
+        if (enableInt8) {
+            auto params = LayerTransformation::Params(
+                true,  // updatePrecisions
+                LayerTransformation::QuantizedTensorAlignment::UpdateLevel,  // quantizedTensorAlignmentOnActivations
+                LayerTransformation::QuantizedTensorAlignment::None,  // quantizedTensorAlignmentOnWeights
+                true);  // supportAsymmetricQuantization
+            LowPrecisionTransformer transformer(LowPrecisionTransformer::getAllTransformations(params)
```

Comment:
@eshoguli, @slyalin, @GlebKazantaev  why the API for low precision transformation is completely differs from common transformations?
We use common transforms as follows:
```cpp
manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
auto pass_config = manager.get_pass_config();
pass_config->disable<ngraph::pass::ConvertGELU>();
pass_config->set_callback<ngraph::pass::ConvertReduceMaxToPooling>(
                [](const_node_ptr &node) -> bool {
                    return disableReduceDecomposition<ngraph::opset1::ReduceMax>(node);
                });
manager.run_passes(nGraphFunc);
// etc
```

So I'd expect LPT to be used as follows:
```cpp
manager.register_pass<ngraph::pass::LowPrecisionTransformations>();
pass_config.set_callback<ngraph::pass:MatMulTransformation>(/* callback that checks asymmetric quantization */);
or 
pass_config.disable<ngraph::pass:MatMulTransformationAsymmetric>();
manager.run_passes(nGraphFunc);
```
It it possible to align LPT with other transforms?